### ISSUE TITLE
Helm ignoring child chart’s values propagation to the parent chart

### ISSUE THREAD

**Issue by ferantivero** [helm/helm#7045]
When a [dependency](https://v2.helm.sh/docs/developing_charts/#chart-dependencies) (`requirements.yaml`) for a chart attempts to [propagate its values (data) to override parent's chart values](https://v2.helm.sh/docs/developing_charts/#importing-child-values-via-requirements-yaml), it results in being ignored.

EDIT: Although reproducible with Helm v2.16.1, I've been to track it down to v.15.0 in which ~- I'm totally guessing -~ it has been introduced by ~https://github.com/helm/helm/pull/6080~ https://github.com/helm/helm/pull/5821 specifically from https://github.com/helm/helm/pull/5821/commits/dda84976bf75e4f3fca9090f0aabe6a7208d709d.

My use case is relatively simple, parent chart can be installed with/without its dependencies using [tags](https://v2.helm.sh/docs/developing_charts/#tags-and-condition-fields-in-requirements-yaml). Therefore, parent chart will provide with some default values for standalone installations but when getting it fully installed with its dep(s), default parent's chart values could be override. It's been working as expected till v2.14.2.

Please let me know if am I missing something around docs/understanding. 

UPDATE: after reverting https://github.com/helm/helm/pull/5821/commits/dda84976bf75e4f3fca9090f0aabe6a7208d709d everything seems to be back to normal

## Repro steps
This can be easily reprod by downloading [foo-0.1.0.zip](https://github.com/helm/helm/files/3876002/foo-0.1.0.zip) + executing the following command or just create your own a chart using the details below:

`helm install foo-0.1.0.zip --dep-up --dry-run --debug`

Expected COMPUTED VALUES (<= v2.14.2; aligned with [docs](https://v2.helm.sh/docs/developing_charts/#importing-child-values-via-requirements-yaml) based on my understanding)

```yaml
...
COMPUTED VALUES:
bar:
  exports:
    data:
      nonce: 42
  global: {}
nonce: 42
...
```

Actual COMPUTED VALUES (>= v2.15.0)

```yaml
...
COMPUTED VALUES:
bar:
  exports:
    data:
      nonce: 42
  global: {}
...
```

Output of `helm version`:

Client: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}

Output of `kubectl version`:

Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:40:16Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"2640ac46d96791a135961425127d9c2f7e184924", GitTreeState:"clean", BuildDate:"2019-11-14T04:58:54Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 
N/A

## Chart details

```
foo
|--- charts
|    |--- bar-0.1.0.tgz
...
|--- deps
|    |--- bar
|    |    |--- Chart.yaml
|    |    |--- values.yaml
...
|--- requirements.yaml
|--- values.yaml
```

```yaml
# ./deps/bar/values.yaml
exports:
  data:
    nonce: 42
```

```yaml
# values.yaml
nonce: null
```

```yaml
# requirements.yaml
dependencies:
  - name: bar
    repository: "file://deps/bar"
    version: ">= 0.0.1"
    import-values:
      - data
```

**Comment by prbsparx** [comment]
Impacted by this as well. Hoping this gets worked into a new version soon.

**Comment by bridgetkromhout** [comment]
Closed in https://github.com/helm/helm/pull/7047 which became available in Helm 2.16.4. Per https://github.com/helm/helm/pull/7047#issuecomment-581608576, this is not a problem in Helm 3.

**Comment by messiahUA** [comment]
Sorry for bringing up this old problem, but I have a situation where this "bug" was used as a feature: child values are imported, but they don't override parent values which makes sense in the current setup. I suppose the most appropriate way to solve this is to provide these values as an input to install/template, but I'm thinking that it's not very logical that dependency has a higher priority and overrides values.

I would appreciate any thoughts, maybe I'm not thinking straight? :)
